test/core/DependencyUnitTests/CMakeLists.txt

```diff
@@ -123,11 +123,11 @@ function(read_expected_output_file_into_regex_array  EXPECTED_OUTPUT_FILE
     #print_var(EXPECTED_RAW_STR)
     if (NOT "${EXPECTED_RAW_STR}" STREQUAL "")
       set(EXPECTED_REGEX_STR "${EXPECTED_RAW_STR}")
-      string(REPLACE "(" "." EXPECTED_REGEX_STR "${EXPECTED_REGEX_STR}") 
-      string(REPLACE ")" "." EXPECTED_REGEX_STR "${EXPECTED_REGEX_STR}") 
-      string(REPLACE "[" "." EXPECTED_REGEX_STR "${EXPECTED_REGEX_STR}") 
-      string(REPLACE "]" "." EXPECTED_REGEX_STR "${EXPECTED_REGEX_STR}") 
-      string(REPLACE "*" "." EXPECTED_REGEX_STR "${EXPECTED_REGEX_STR}") 
+      string(REPLACE "[" "." EXPECTED_REGEX_STR "${EXPECTED_REGEX_STR}")
+      string(REPLACE "]" "." EXPECTED_REGEX_STR "${EXPECTED_REGEX_STR}")
+      string(REPLACE "(" "[(]" EXPECTED_REGEX_STR "${EXPECTED_REGEX_STR}")
```

Comment:
I have not had good luck using `\\` to try to escape chars.  That is because it seems like one or more `\` will get removed as you pass arguments through layers of functions.  This behavior confuses me to the point that it is just not worth the tiny bit of extra checking in the output for differentiate a `[` from some arbitrary char compared to fighting with how many `\` are needed for a given situation.